File: Friends/Friends.Common/Application/UsesCases/GetFriendUseCase.cs
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using AutoMapper;
using Friends.Common.Domain.Interfaces;
using Friends.Common.UI.Models;

namespace Friends.Common.Application.UsesCases
{
	public class GetFriendUseCase
	{
        private readonly IFriendRepository _friendRepository;
        private readonly IFriendCacheRepository _friendCacheRepository;
        private readonly IMapper _mapper;

        public GetFriendUseCase(IFriendRepository friendRepository, IFriendCacheRepository friendCacheRepository, IMapper mapper)
		{
			_friendRepository = friendRepository;
            _friendCacheRepository = friendCacheRepository;
            _mapper = mapper;
        }

		public async Task<List<FriendUI>> ExecuteAsync(CancellationToken ct = default)
		{
			var friendList = await _friendRepository.GetFriendsAsync(ct);
            _friendCacheRepository.SaveFriends(friendList);
            var friendUIList = _mapper.Map<List<FriendUI>>(friendList);
            return friendUIList;
        }
	}
}

File: Friends.Kotlin/app/src/main/java/com/friends/presentation/usecases/GetFriendUseCase.kt
package com.friends.presentation.usecases

import com.friends.domain.interfaces.IFriendsRepository
import com.friends.presentation.mapper.toFriendUiList
import com.friends.presentation.models.FriendUi
import com.friends.presentation.ui.states.FriendStates
import javax.inject.Inject


class GetFriendUseCase @Inject constructor(
    private val friendsRepository: IFriendsRepository
){
    suspend fun executeAsync(): List<FriendUi> {
        val friendList = friendsRepository.getFriendsAsync()
        val friendUiList = friendList.toFriendUiList()
        FriendStates.friends = friendUiList
        return friendUiList
    }
}